webknossos/webknossos/dataset/view.py

```diff
@@ -139,40 +281,130 @@ def read(
         ```
         """
 
-        offset = Vec3Int(offset)
-        size = self.size if size is None else Vec3Int(size)
+        current_mag_size: Optional[Vec3IntLike]
+        mag1_size: Optional[Vec3IntLike]
+        if absolute_bounding_box is None and relative_bounding_box is None:
+            if offset is None:
+                if size is None:
+                    assert (
+                        relative_offset is None and absolute_offset is None
+                    ), "You must supply size, when reading with an offset."
+                    current_mag_size = None
+                    mag1_size = self.bounding_box.size
+                else:
+                    if relative_offset is None and absolute_offset is None:
+                        if type(self) == View:
+                            offset_param = "relative_offset"
+                        else:
+                            offset_param = "absolute_offset"
+                        warnings.warn(
+                            "[DEPRECATION] Using view.read(size=my_vec) only with a size is deprecated. "
+                            + f"Please use view.read({offset_param}=(0, 0, 0), size=size_vec * view.mag.to_vec3_int()) instead."
+                        )
+                        current_mag_size = size
+                        mag1_size = None
+                    else:
+                        current_mag_size = None
```

Comment:
current_mag_size is set to `None` in all branches of the level-2 if-block (`if offset is None`). I'd move it to the end of the if-block to deduplicate it.

Reply:
There's still `current_mag_size = size`, so I don't think that's possible. `mag1_size` and `current_mag_size` are always set in pairs, since exactly one of them should be used.

Reply:
Whoops, you are right. Nevermind then.